nonce/nonce.go

```diff
@@ -117,11 +131,25 @@ func (ns *NonceService) encrypt(counter int64) (string, error) {
 	ct := ns.gcm.Seal(nil, nonce, pt, nil)
 	copy(ret, nonce[4:])
 	copy(ret[8:], ct)
-	return base64.RawURLEncoding.EncodeToString(ret), nil
+
+	return ns.prefix + base64.RawURLEncoding.EncodeToString(ret), nil
 }
 
 func (ns *NonceService) decrypt(nonce string) (int64, error) {
-	decoded, err := base64.RawURLEncoding.DecodeString(nonce)
+	var decoded []byte
+	if ns.prefix != "" {
+		var prefix string
+		var err error
+		prefix, nonce, err = splitNonce(nonce)
+		if err != nil {
+			return 0, err
+		}
+		if ns.prefix != prefix {
+			return 0, errors.New("nonce contains invalid prefix")
```

Comment:
I feel like including the actual vs expected nonce prefix in the err message returned here might be valuable in the future if we need to debug nonce redemption issues.

I don't feel super strongly about this if you disagree. Feel free to ignore,

```suggestion
			return 0, fmt.Errorf("nonce contains invalid prefix %q vs %q", prefix, ns.prefix)
```